src/solver/generic.ml

```diff
@@ -63,21 +63,21 @@ struct
 
   let new_var_event x =
     incr SolverStats.vars;
-    if tracing then trace "sol" "New %a\n" Var.pretty_trace x
+    if tracing then trace "sol" "New %a" Var.pretty_trace x
 
   let get_var_event x =
-    if full_trace then trace "sol" "Querying %a\n" Var.pretty_trace x
+    if full_trace then trace "sol" "Querying %a" Var.pretty_trace x
 
   let eval_rhs_event x =
-    if full_trace then trace "sol" "(Re-)evaluating %a\n" Var.pretty_trace x;
+    if full_trace then trace "sol" "(Re-)evaluating %a" Var.pretty_trace x;
     incr SolverStats.evals;
     if (get_bool "dbg.solver-progress") then (incr stack_d; Logs.debug "%d" !stack_d)
 
   let update_var_event x o n =
     if tracing then increase x;
     if full_trace || ((not (Dom.is_bot o)) && Option.is_some !max_var && Var.equal (Option.get !max_var) x) then begin
-      if tracing then tracei "sol_max" "(%d) Update to %a.\n" !max_c Var.pretty_trace x;
-      if tracing then traceu "sol_max" "%a\n\n" Dom.pretty_diff (n, o)
+      if tracing then tracei "sol_max" "(%d) Update to %a" !max_c Var.pretty_trace x;
+      if tracing then traceu "sol_max" "%a" Dom.pretty_diff (n, o)
```

Comment:
`full_trace` and `trace` are unused inconsistently above and here. Should `full_trace` on its cause tracing?

Reply:
I don't think so, because in different places `full_trace` is used both positively and negatively to do `trace` calls, so it's intended to be some kind of switch.
I made a few calls also conditional on `tracing`, so that controls everything like it should be. `full_trace` just changes what's being traced (if anything).